ignite/handlers/param_scheduler.py

```diff
@@ -313,6 +315,18 @@ def __init__(
         self.cycle = 0
         self.start_value_mult = start_value_mult
         self.end_value_mult = end_value_mult
+        self.warmup_each_cycle = warmup_each_cycle
+        if not self.warmup_each_cycle:
+            if warmup_duration is not None:
+                warnings.warn(
```

Comment:
We should add proper tests to cover the new changes. For this uncovered line and also line 367, you could add this test to `tests/ignite/handlers/test_param_scheduler.py`:
```python
def test_cyclical_scheduler_asserts():
    tensor = torch.zeros([1], requires_grad=True)
    optimizer = torch.optim.SGD([tensor], lr=0)
    with pytest.warns(match="warmup_each_cycle=False but your warmup_duration"):
        scheduler = CyclicalScheduler(optimizer, "lr", 0., 1., 10, warmup_each_cycle=False, warmup_duration=2)

    with pytest.raises(NotImplementedError):
       scheduler._get_cycle_param()
```